### DISCUSSION TITLE
unable to install truffle

### DISCUSSION THREAD

**Discussion by samdupnamgyal** [hashlips-lab/nft-erc721-collection#260] [Q&A]
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/truffle
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/truffle'
npm ERR! [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/truffle'] {
npm ERR! errno: -13,
npm ERR! code: 'EACCES',
npm ERR! syscall: 'mkdir',
npm ERR! path: '/usr/local/lib/node_modules/truffle'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR! /Users/sam/.npm/_logs/2022-04-24T13_42_37_145Z-debug-0.log
sam@SAMs-MacBook-Pro nft-erc721-collection-2.3.0 % root/Administrator
zsh: no such file or directory: root/Administrator
sam@SAMs-MacBook-Pro nft-erc721-collection-2.3.0 %

**Comment by liarco** [comment]
Hi @samdupnamgyal, please take a look at my answer here: https://github.com/hashlips-lab/nft-erc721-collection/issues/259#issuecomment-1107848124

If you installed NodeJS with another method instead of the one that I show in my video you may have a permissions issue. In that case I think the fastest solution would be running the command with `sudo` (e.g. `sudo npm i truffle` and entering your password), but please keep in mind that this is not a best practice since it runs the command as an Administrator on your machine. **Use it at your own risk.**

**Reply by Semikrc** [reply]
I am a new member of github community. And also I have designer and create nfts. I watch all of your videos but I have an error and I couldnt find a solitioun. And also I really dont know create a discussion part because as you see my english skills not goot. Please help me about my error. I have an error like this;

node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v83-win32-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for sqlite3@4.2.0 and node@14.18.2 (node-v83 ABI, unknown) (falling back to source compile with node-gyp)
Bu çözümde projeler tek tek derlenir. Paralel bir derlemeyi etkinleştirmek için lütfen "/m" anahtarını ekleyin.
  unpack_sqlite_dep
  Python bulunamad
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.e
xe" öğesinden 9009 koduyla çıkıldı. [C:\Users\semik\AppData\Roaming\npm\node_modules\truffle\node_modules\sqlite3\build\deps\action_before_build. 
vcxproj]
gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1      
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
gyp ERR! System Windows_NT 10.0.22598
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "build" "--fallback-to-build" "--module=C:\\Users\\semik\\AppData\\Roaming\\npm\\node_modules\\truffle\\node_modules\\sqlite3\\lib\\binding\\node-v83-win32-x64\\node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=C:\\Users\\semik\\AppData\\Roaming\\npm\\node_modules\\truffle\\node_modules\\sqlite3\\lib\\binding\\node-v83-win32-x64" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v83"
gyp ERR! cwd C:\Users\semik\AppData\Roaming\npm\node_modules\truffle\node_modules\sqlite3
gyp ERR! node -v v14.18.2
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\Users\semik\AppData\Roaming\npm\node_modules\truffle\node_modules\sqlite3\lib\binding\node-v83-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\semik\AppData\Roaming\npm\node_modules\truffle\node_modules\sqlite3\lib\binding\node-v83-win32-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v83' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\semik\AppData\Roaming\npm\node_modules\truffle\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1058:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
node-pre-gyp ERR! System Windows_NT 10.0.22598
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\semik\\AppData\\Roaming\\npm\\node_modules\\truffle\\node_modules\\nodnpm WARN enoent SKIPPING OPTIONAL DEPENDENCY:
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: spawn git ENOENT
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules\truffle\node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sqlite3@4.2.0 (node_modules\truffle\node_modules\sqlite3):npm WARN optional SKIPPING OPTIONAL DEPENDENCY: sqlite3@4.2.0 install: `node-pre-gyp install --fallback-to-build`npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1
